meerk40t/gui/spoolerpanel.py

```diff
@@ -743,235 +654,64 @@ def datestr(t):
+            self.list_job_history.SetItem(list_id, HC_STATUS, info.get("status", ""))
+            self.list_job_history.SetItem(list_id, HC_JOBINFO, info.get("info", ""))
             self.list_job_history.SetItem(
-                list_id, self.column_history["runtime"], runtime
+                list_id,
+                HC_ESTIMATE,
+                self.timestr(info.get("estimate", 0), False),
             )
-            # First passes then device
-            if len(info) >= 5:
-                self.list_job_history.SetItem(
-                    list_id,
-                    self.column_history["passes"],
-                    str(info[IDX_HISTORY_PASSES]),
-                )
-            else:
-                self.list_job_history.SetItem(
-                    list_id, self.column_history["passes"], "???"
-                )
             self.list_job_history.SetItem(
-                list_id, self.column_history["device"], str(info[IDX_HISTORY_DEVICE])
+                list_id,
+                HC_STEPS,
+                f"{info.get('steps_done',0)}/{info.get('steps_total',0)}",
             )
-            if len(info) > IDX_HISTORY_STATUS:
-                if info[IDX_HISTORY_INFO] is None:
-                    info[IDX_HISTORY_INFO] = ""
-                self.list_job_history.SetItem(
-                    list_id, self.column_history["status"], info[IDX_HISTORY_STATUS]
-                )
-            if len(info) > IDX_HISTORY_INFO:
-                if isinstance(info[IDX_HISTORY_INFO], bool):
-                    # Old data, where helper flag was saved
-                    info[IDX_HISTORY_INFO] = ""
-                if info[IDX_HISTORY_INFO] is None:
-                    info[IDX_HISTORY_INFO] = ""
-                self.list_job_history.SetItem(
-                    list_id, self.column_history["jobinfo"], info[IDX_HISTORY_INFO]
-                )
-            if len(info) > IDX_HISTORY_ESTIMATE:
-                if info[IDX_HISTORY_ESTIMATE] is None:
-                    info[IDX_HISTORY_ESTIMATE] = 0
-                runtime = self.timestr(info[IDX_HISTORY_ESTIMATE], False)
-                self.list_job_history.SetItem(
-                    list_id, self.column_history["estimate"], runtime
-                )
-            if len(info) > IDX_HISTORY_STEPS_TOTAL:
-                if info[IDX_HISTORY_STEPS_DONE] is None:
-                    info[IDX_HISTORY_STEPS_DONE] = 0
-                if info[IDX_HISTORY_STEPS_TOTAL] is None:
-                    info[IDX_HISTORY_STEPS_TOTAL] = 0
-                stepinfo = (
-                    f"{info[IDX_HISTORY_STEPS_DONE]}/{info[IDX_HISTORY_STEPS_TOTAL]}"
-                )
-                self.list_job_history.SetItem(
-                    list_id, self.column_history["steps"], stepinfo
-                )
-            self.list_job_history.SetItemData(list_id, idx - 1)
+            self.list_job_history.SetItemData(list_id, idx)
 
     def reload_history(self):
-        def fixitems():
```

Comment:
I was able to crash meerk40t consistently with an improperly stored logging string, part of this routine is not only to get backward compatibility (which is no longer an issue) but to deal with malformed entries. So the:
```
           if isinstance(self.history, str):
                # Flawed !! At least recover gracefully
                self.history = []
```
should be reestablished